### PR TITLE
refactor(common/lmlayer): create package: @keymanapp/models-templates

### PR DESCRIPTION
**Pull request by eddieantonio**
This creates a package to store all model backends. Currently, that's only the `TrieModel`!

This demonstrates a few things:

 - a conventional TypeScript/NodeJS package organization
 - use npm scripts to good use, like `prepare` to build the TypeScript before the package is bootstrapped into other projects.
 - the fact that namespace packages are a nightmare. We're refactoring to ES6 modules in the next Keyman version, right?

Currently a draft, as I want @jahorton's 👁️  on this before I do anything significant.

### CODE REVIEW

**Comment by jahorton** [comment]
After some recent discussion on nomenclature, I believe we'll be going with `@keymanapp/models-templates` now, in the folder `common/models/templates`.  

Aside from that... we've got a number of other moving pieces in the same area at the moment... I'd like to wait for a couple of them to resolve first before going in-depth here.

File: common/predictive-text/package.json [hunk 1]
```diff
@@ -45,6 +45,9 @@
     "typescript": "^3.7.2"
   },
   "dependencies": {
+    "@keymanapp/keyboard-processor": "^14.0.76",
```
Comment (jahorton) [inline common/predictive-text/package.json:48]: Why is `keyboard-processor` a dependency here?  `web-utils` was created explicitly to avoid this.

**Comment by jahorton** [comment]
This should probably be reviewed / merged after https://github.com/keymanapp/keyman/pull/3147 is merged to `master`.  In `master`'s current state, we're using both the now-deprecated `@keymanapp/lexical-model-types` and `@keymanapp/model-types`, which is causing some local build instabilities.  (When the former was also in place, `lerna bootstrap` was falling over due to duplicated type definitions in some of the new `prepublish` steps!)

I want to be sure that this doesn't break predictive text before fully reviewing merging.  We're probably fine, but it's still best to do an actual build to check.

File: common/predictive-text/package.json (hunk 1 above)
Comment (eddieantonio) [reply common/predictive-text/package.json:48]: Must have gotten in there by mistake (somehow????). Nice catch!

**Comment by eddieantonio** [comment]
Alright! I merged master, and now I'm just waiting for TeamCity 🤞

**Comment by eddieantonio** [comment]
The only failed TeamCity build failed on git checkout?????

> ```
> Failed to perform checkout on agent: '"C:\Program Files\Git\bin\git.exe" clean -f -d -x' command failed.
> ```

**Comment by mcdurdin** [comment]
> The only failed TeamCity build failed on git checkout?????
> 

```
stderr: warning: could not open directory 'common/core/web/input-processor/node_modules/@keymanapp/lexical-model-layer/node_modules/@keymanapp/models-templates/node_modules/@keymanapp/models-wordbreakers/node_modules/@keymanapp/models-types/node_modules/typescript/': Filename too long
```

Another build bites the dust. Fixing manually and re-triggering the build. This will soon go away once we have no more open PRs that trigger the issue. (It really sucks when environmental factors caused by other builds cause otherwise good little builds to fail).

**Pull request merged**